solidity/contracts/bridge/Fraud.sol

```diff
@@ -109,32 +109,41 @@ library Fraud {
     ///         challenge or confiscated otherwise
     /// @param walletPublicKey The public key of the wallet in the uncompressed
     ///        and unprefixed format (64 bytes)
-    /// @param sighash The hash that was used to produce the ECDSA signature
-    ///        that is the subject of the fraud claim. This hash is constructed
-    ///        by applying double SHA-256 over a serialized subset of the
-    ///        transaction. The exact subset used as hash preimage depends on
-    ///        the transaction input the signature is produced for. See BIP-143
-    ///        for reference
+    /// @param sha256preimage The hash that was generated by applying SHA-256
+    ///        one time over the preimage used during input signing. The preimage
+    ///        is a serialized subset of the transaction and its structure
+    ///        depends on the transaction input (see BIP-143 for reference).
+    ///        Notice that applying SHA-256 over the `sha256preimage` results
+    ///        in `sighash`.  The path from `preimage` to `sighash` looks like
+    ///        this:
+    ///        preimage -> (SHA-256) -> sha256preimage -> (SHA-256) -> sighash
     /// @param signature Bitcoin signature in the R/S/V format
     /// @dev Requirements:
     ///      - Wallet behind `walletPublicKey` must be in Live or MovingFunds
     ///        or Closing state
     ///      - The challenger must send appropriate amount of ETH used as
     ///        fraud challenge deposit
     ///      - The signature (represented by r, s and v) must be generated by
-    ///        the wallet behind `walletPublicKey` during signing of `sighash`
+    ///        the wallet behind `walletPubKey` during signing of `sighash`
+    ///        which was calculated from `sha256preimage`
     ///      - Wallet can be challenged for the given signature only once
     function submitFraudChallenge(
         BridgeState.Storage storage self,
         bytes calldata walletPublicKey,
-        bytes32 sighash,
+        bytes memory sha256preimage,
```

Comment:
I'd name it `preimageSha256` because it will show the order of operations in a better way (preimage -> sha256).

Reply:
Agreed.

Reply:
Done in 23a59f3e45809064c84f9678bc96e60529bec98e.